crates/router/src/connector/multisafepay/transformers.rs

```diff
@@ -0,0 +1,453 @@
+use std::fmt::format;
+
+use api_models::payments::Card;
+use common_utils::pii::Email;
+use error_stack::ResultExt;
+use masking::{PeekInterface, ExposeInterface};
+use regex::Error;
+use serde::{Deserialize, Serialize};
+use url::Url;
+use crate::{core::errors,pii::{self, Secret},services,types::{self,api, storage::enums}, utils::OptionExt};
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum Gateway {
+    Amex,
+    Creditcard,
+    Maestro,
+    Mastercard,
+    Visa
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+pub struct Coupons {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub allow: Option<Vec<String>>
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+pub struct Mistercash {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub mobile_pay_button_position: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub disable_mobile_pay_button: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub qr_only: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub qr_size: Option<String>
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "UPPERCASE")]
+pub struct Gateways {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub mistercash: Option<Mistercash>
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+pub struct Settings {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub coupons: Option<Coupons>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub gateways: Option<Gateways>
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+pub struct PaymentOptions {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub notification_url: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub notification_method: Option<String>, 
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub redirect_url: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub cancel_url: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub close_window: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub settings: Option<Settings>, 
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub template_id: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    // pub template: Option<Template>,
+    pub allowed_countries: Option<Vec<String>>
+
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+pub struct Browser {
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
Does the payment gateway return error if null is present in this field? I'm trying to understand the reason for skipping serialisation for all the fields if none